src/linear_samplers/block_col_rand_cyclic.jl

```diff
@@ -0,0 +1,92 @@
+# This code was written by Nathaniel Pritchard
+"""
+    LinSysBlkColRandCyclic <: LinSysBlkColSampler 
+
+A mutable structure with fields to handle randomly permuted block sampling. Can allow for fixed
+blocks or blocks whose entries are randomly permuted. After each cycle, a new random ordering is
+created. If the last block would be smaller than the others, columns from a previous block are 
+added to keep all blocks the same size.
```

Comment:
Is this for memory utilization purposes?

Reply:
For both those purposes and code simplicity. Without doing this, as I have in the past. It is necessary to define a views for the blocks and residual vectors that will have to be computed. Additionally, extra conditional statements need to be added to account for this case. Overall, it is far more complicated than would be desired if this is not done.

Reply:
I think this is an optimization that precludes the user from designing blocks as they see fit. This function should let the user specify which columns to include in which blocks if they want to. There can be a constructor that allows for the blocks to be "uniform" in size as you have specified here to simplify the construction.

Reply:
I changed it to allow the user to input a vector of vectors.